From akpm@osdl.org Wed Nov 30 15:21:12 2005
Message-Id: <200511302319.jAUNJnoO015372@shell0.pdx.osdl.net>
From: Pekka Enberg <penberg@cs.helsinki.fi>
Subject: aoe: skb_check cleanup
To: penberg@cs.helsinki.fi, axboe@suse.de, greg@kroah.com, jgarzik@pobox.com
Date: Wed, 30 Nov 2005 15:21:08 -0800

This patch cleans up skb_check() to make it readable.

Signed-off-by: Pekka Enberg <penberg@cs.helsinki.fi>
Cc: Jeff Garzik <jgarzik@pobox.com>
Cc: Jens Axboe <axboe@suse.de>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/block/aoe/aoenet.c |   11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

--- gregkh-2.6.orig/drivers/block/aoe/aoenet.c
+++ gregkh-2.6/drivers/block/aoe/aoenet.c
@@ -95,11 +95,12 @@ mac_addr(char addr[6])
 static struct sk_buff *
 skb_check(struct sk_buff *skb)
 {
-	if (skb_is_nonlinear(skb))
-	if ((skb = skb_share_check(skb, GFP_ATOMIC)))
-	if (skb_linearize(skb, GFP_ATOMIC) < 0) {
-		dev_kfree_skb(skb);
-		return NULL;
+	if (skb_is_nonlinear(skb)) {
+		skb = skb_share_check(skb, GFP_ATOMIC);
+		if (skb && skb_linearize(skb, GFP_ATOMIC) < 0) {
+			dev_kfree_skb(skb);
+			return NULL;
+		}
 	}
 	return skb;
 }
